pkg/controller/external/util.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package external
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Get uses the client and reference to get an external, unstructured object.
+func Get(c client.Client, ref *corev1.ObjectReference, namespace string) (*unstructured.Unstructured, error) {
+	obj := new(unstructured.Unstructured)
+	obj.SetAPIVersion(ref.APIVersion)
+	obj.SetKind(ref.Kind)
+	obj.SetName(ref.Name)
+	key := client.ObjectKey{Name: obj.GetName(), Namespace: namespace}
+	if err := c.Get(context.Background(), key, obj); err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+// Clone uses the client and the reference to create a new cloned object.
+func Clone(c client.Client, ref *corev1.ObjectReference, namespace string) (*unstructured.Unstructured, error) {
+	from, err := Get(c, ref, namespace)
+	if err != nil {
+		return nil, err
+	}
+	to := from.DeepCopy()
+	to.SetResourceVersion("")
+	to.SetOwnerReferences(nil)
+	to.SetName("")
```

Comment:
@detiber Can you quickly double check that these are the only things we should reset on an object before creating a new copy of it?

Reply:
Would we also potentially need to worry about finalizers here as well? @ncdc

Reply:
I'm 50-50 on this. The end result is probably the same either way. Also, I don't see this being called yet?

Reply:
Not yet, I followed the code from the pivot. I added a few more resets, just in case